src/Serialization.cpp

```diff
@@ -127,7 +127,7 @@ class Serializer {
 
     Offset<Serialize::ExternFuncArgument> serialize_extern_func_argument(FlatBufferBuilder &builder, const ExternFuncArgument &extern_func_argument);
 
-    Offset<Serialize::Buffer> serialize_buffer(FlatBufferBuilder &builder, const Buffer<> &buffer);
+    Offset<Serialize::Buffer> serialize_buffer(FlatBufferBuilder &builder, Buffer<> &buffer);
```

Comment:
A Buffer is already a pointer type. If you want to mutate it, just accept it by value. In general we should avoid passing things by non-const reference.